examples/chat.rs

```diff
@@ -133,50 +133,47 @@ async fn main() -> Result<(), Box<dyn Error>> {
     }
 
     // Read full lines from stdin
-    let mut stdin = io::BufReader::new(io::stdin()).lines();
+    let mut stdin = io::BufReader::new(io::stdin()).lines().fuse();
 
     // Listen on all interfaces and whatever port the OS assigns
     swarm.listen_on("/ip4/0.0.0.0/tcp/0".parse()?)?;
 
     // Kick it off
-    task::block_on(future::poll_fn(move |cx: &mut Context<'_>| {
-        loop {
-            match stdin.try_poll_next_unpin(cx)? {
-                Poll::Ready(Some(line)) => swarm
+    loop {
+        select! {
+            line = stdin.next() => match line {
+                Some(line) => swarm
```

Comment:
```suggestion
            line = stdin.next() => swarm.behaviour_mut().floodsub.publish(floodsub_topic.clone(), line.expect("Stdin not to close")?.as_bytes())
```

What do you think? Should be a bit simpler?

Reply:
Yeah, I agree. Getting rid of the other branch does simplify the code. After all, these are examples, there is no need for thorough error handling.

Reply:
I will apply these changes and test them briefly.